### PR TITLE
[Doc] Add Schema Evolution and Compatibility section

### PR DESCRIPTION
**Pull request by Anonymitaet**
Structure of Schema Chapter: https://github.com/apache/pulsar/issues/4789

### CODE REVIEW

File: site2/docs/schema-evolution-compatibility.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+Schemas store the details of attributes and types. To satisfy new business requirements,  you need to update schemas inevitably over time, which is called **schema evolution**. 
+
+Any schema changes affect downstream consumers. Schema evolution ensures that the downstream consumers can seamlessly handle data encoded with both old schemas and new schemas. 
+
+### How Pulsar schema should evolve? 
+
+The answer is Pulsar schema compatibility check strategy. It determines how schema compares old schemas with new schemas in topics.
+
+For more information, see [Schema compatibility check strategy](#schema-compatibility-check-strategy).
+
```
Comment (Anonymitaet) [inline site2/docs/schema-evolution-compatibility.md]: @sijie Line 15-25 are my writings, are they correct?

File: site2/docs/schema-evolution-compatibility.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+| `ALWAYS_INCOMPATIBLE` | Disable schema evolution | All changes are disabled | / | / |
+| `BACKWARD` | Consumers using the schema V3 can process data written by producers using the schema V3 or V2 | Add optional fields<br>Delete fields | Latest version | Consumers |
+| `BACKWARD_TRANSITIVE` | Consumers using the schema V3 can process data written by producers using the schema V3, V2 or V1 | Add optional fields<br>Delete fields | All previous versions | Consumers |
+| `FORWARD` | Consumers using the schema V3 or V2 can process data written by producers using the schema V3 | Add fields<br>Delete optional fields | Latest version | Producers |
+| `FORWARD_TRANSITIVE` | Consumers using the schema V3, V2 or V1 can process data written by producers using the schema V3 | Add fields<br>Delete optional fields | All previous versions | Producers |
+| `FULL` | Backward and forward compatible between the schema V3 and V2 | Modify optional fields | Latest version | Any order |
+| `FULL_TRANSITIVE` | Backward and forward compatible among the schema V3, V2, and V1 | Modify optional fields | All previous versions | Any order |
+
```
Comment (Anonymitaet) [inline site2/docs/schema-evolution-compatibility.md]: @sijie For this table, I add the **Definition** column, are the contents accurate?

File: site2/docs/schema-evolution-compatibility.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+### BACKWARD and BACKWARD_TRANSITIVE 
+
+Suppose that you have a topic containing three schemas (V1, V2, and V3), V1 is the oldest and V3 is the latest:
+
+| Compatibility check strategy | Definition  | Description |
+|---|---|---|
+`BACKWARD` | Consumers using the new schema can process data written by producers using the **last schema**. | The consumers using the schema V3 can process data written by producers using the schema V3 or V2. |
+`BACKWARD_TRANSITIVE` | Consumers using the new schema can process data written by producers using **all previous schemas**. | The consumers using the schema V3 can process data written by producers using the schema V3, V2, or V1. |
+
```
Comment (Anonymitaet) [inline site2/docs/schema-evolution-compatibility.md]: @sijie I re-write this table, is it accurate?

File: site2/docs/schema-evolution-compatibility.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+### FORWARD and FORWARD_TRANSITIVE 
+
+Suppose that you have a topic containing three schemas (V1, V2, and V3), V1 is the oldest and V3 is the latest:
+
+| Compatibility check strategy | Definition | Description |
+|---|---|---|
+`FORWARD` | Consumers using the **last schema** can process data written by producers using a new schema, even though they may not be able to use the full capabilities of the new schema. | The consumers using the schema V3 or V2 can process data written by producers using the schema V3. |
+`FORWARD_TRANSITIVE` | Consumers using **all previous schemas** can process data written by producers using a new schema. | The consumers using the schema V3, V2, or V1 can process data written by producers using the schema V3. 
+
```
Comment (Anonymitaet) [inline site2/docs/schema-evolution-compatibility.md]: @sijie I re-write this table, is it accurate?

File: site2/docs/schema-evolution-compatibility.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+| Compatibility check strategy | Definition | Description | Note |
+|---|---|---|---|
+`FULL` |  Schemas are both backward and forward compatible, which means consumers using the last schema can process data written by producers using the new schema and consumers using the new schema can process data written by producers using the last schema. | The consumers using the schema V3 can process data written by producers using the schema V3 or V2, and the consumers using the schema V3 or V2 can process data written by producers using the schema V3. | For Avro and JSON, the default schema compatibility check strategy is `FULL`. <br>For all schema types except Avro and JSON, the default schema compatibility check strategy is `ALWAYS_INCOMPATIBLE`. |
+`FULL_TRANSITIVE` | The new schema is backward and forward compatible with all previously registered schemas. | The consumers using the schema V3 can process data written by the producers using the schema V3, V2 or V1, and the consumers using the schema V3, V2 or V1 can process data written by the producers using the schema V3. |
+
```
Comment (Anonymitaet) [inline site2/docs/schema-evolution-compatibility.md]: @sijie I re-write this table, is it accurate?

File: site2/docs/schema-evolution-compatibility.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+| Compatibility check strategy | Description |
+|---|---|
+`ALWAYS_COMPATIBLE` | The compatibility check is disabled. <br>Consequently, you can upgrade the producers and consumers in **any order**. |
+`ALWAYS_INCOMPATIBLE` | / |
+`BACKWARD`<br>`BACKWARD_TRANSITIVE` | There is no guarantee that consumers using the old schema can read data produced using the new schema. <br>Consequently, **upgrade all consumers first**, and then start producing new data.
+`FORWARD`<br>`FORWARD_TRANSITIVE` | There is no guarantee that consumers using the new schema can read data produced using the old schema. <br>Consequently, **upgrade all producers first** to use the new schema and ensure that the data already produced using the old schemas are not available to consumers, and then upgrade the consumers. |
+`FULL`<br>`FULL_TRANSITIVE` | There is no guarantee that consumers using the old schema can read data produced using the new schema and consumers using the new schema can read data produced using the old schema. <br>Consequently, you can upgrade the producers and consumers in **any order**. |
+
```
Comment (Anonymitaet) [inline site2/docs/schema-evolution-compatibility.md]: @sijie I re-write this table, is it accurate?

File: site2/docs/schema-evolution-compatibility.md (hunk 2 above)
Comment (sijie) [reply site2/docs/schema-evolution-compatibility.md]: looks good to me.

File: site2/docs/schema-evolution-compatibility.md (hunk 5 above)
Comment (sijie) [reply site2/docs/schema-evolution-compatibility.md]: lgtm

File: site2/docs/schema-evolution-compatibility.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+## Order of upgrading clients
+
+The order of upgrading client applications is determined by the compatibility check strategy.
+
+For example, the producers using schemas to write data to Pulsar and the consumers using schemas to read data from Pulsar. 
+
+| Compatibility check strategy | Description |
+|---|---|
+`ALWAYS_COMPATIBLE` | The compatibility check is disabled. <br>Consequently, you can upgrade the producers and consumers in **any order**. |
+`ALWAYS_INCOMPATIBLE` | / |
```
Comment (sijie) [inline site2/docs/schema-evolution-compatibility.md]: "/" doesn't provide useful information. And it sometimes causes confusion. I would suggest adding at least one sentence for it.

**Comment by sijie** [comment]
run integration tests

File: site2/docs/schema-evolution-compatibility.md (hunk 7 above)
Comment (Anonymitaet) [reply site2/docs/schema-evolution-compatibility.md]: @sijie Thank you, I've replaced "/" with "contents".

**Comment by Anonymitaet** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Pull request merged**